packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -258,7 +263,25 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
     if (resultMap.containsKey(requestCode)) {
       final Result result = resultMap.get(requestCode);
       if (requestCode == ACTIVITY_RESULT_REQUEST_CODE_ADD_NETWORKS) {
-        result.success(resultCode == Activity.RESULT_OK);
+        if ((data != null) && data.hasExtra(android.provider.Settings.EXTRA_WIFI_NETWORK_RESULT_LIST)) {
```

Comment:
There might be a problem that a result other than 0, 1, 2 might be returned (error or future addition). In this case the return method would not be triggered. So I would suggest the following:

```dart
int resultValue = -1;
if (data != null && data.hasExtra(EXTRA_WIFI_NETWORK_RESULT_LIST)) {
  for (int code : data.getIntegerArrayListExtra(EXTRA_WIFI_NETWORK_RESULT_LIST)) {
    if (code == ADD_WIFI_RESULT_SUCCESS) {
      resultValue = WIFI_REGISTER_NETWORK_RESULT_SUCCESS;
    } else if (code == ADD_WIFI_RESULT_FAILED) {
      resultValue = WIFI_REGISTER_NETWORK_RESULT_FAILED;
    } else if (code == android.provider.Settings.ADD_WIFI_RESULT_ALREADY_EXISTS) {
      resultValue = WIFI_REGISTER_NETWORK_RESULT_ALREADY_EXISTS;
    }
    break;
  }
}
if (resultCode != -1) {
  result.error("Error", "Unknown return value for ADD_WIFI_RESULT", "");
} else {
  result.success(resultCode);
}
```

One might also import android.provider.Settings

Reply:
Fixed